src/EventTickets/DataTransferObjects/TicketPurchaseData.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace Give\EventTickets\DataTransferObjects;
+
+use Give\EventTickets\Models\EventTicketType;
+use stdClass;
+
+/**
+ * @unreleased
+ */
+class TicketPurchaseData
+{
+    /**
+     * @unreleased
+     * @var int
+     */
+    protected $quantity;
+
+    /**
+     * @unreleased
+     * @var EventTicketType
+     */
+    protected $ticketType;
+
+    /**
+     * @unreleased
+     */
+    public function __get($name)
+    {
+        return $this->$name;
+    }
+
+    /**
+     * @unreleased
+     */
+    public static function fromFieldValueObject(stdClass $object): self
+    {
+        $self = new self();
+
+        $self->quantity = $object->quantity;
+        $self->ticketType = EventTicketType::find($object->ticketId);
```

Comment:
In case you think it's worth to be added, we also have the `amount` attribute being passed. It's `quantity * ticketPrice` and it's passed as minor amount.

Reply:
I don't think we need to use the amount here and I would be cautious to use numerical values passed by the client which can be altered.

Reply:
Right! I've noticed the difference in my approach here compared to the Fee Recovery feature. For Fee Recovery, I introduced a custom validation rule to detect any modifications made by the user in the frontend. This backend validation ensures any changes are caught, preventing the processing of the donation if discrepancies are found. This method enhances our certainty level at this point. However, without such validation, we would need to re-calculate using backend data to ensure it's correct instead.

Reply:
Here I think we just need to re-fetch the ticket amount from the database, which would prevent any price tampering from the client side.